### PR TITLE
Fixed Bugs in the code and instructions in the User Guide

### PR DESCRIPTION
**Pull request by SalsabilTasnia**
Resolved issues #168  #163 #158 #154 #152 #151 #147 #126 
For issue #156 I have updated the user guide on the usage of the View all menu command
Made changes to the User Guide suggested based on the previous pr review

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -89,19 +89,28 @@ Format: `listmenu`
...
+
 ==== View All Menu
 
 Shows the detail of the a menu item with a particular index. +
 Format: `viewallmenu INDEX`
 
+Example: +
+listmenutype main +
+viewallmenu 1 +
+System shows all the main dishes in the menu.
```
Comment (px1099) [inline docs/UserGuide.adoc]: Line 112 seems to be incorrect

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -134,9 +143,12 @@ Format: `deletemenu INDEX`
 * Deletes the menu item in that particular index
 
 Example: +
-viewallmenu +
+listmenutype main +
 deletemenu 1 +
 System deletes food item in index 1 from the menu. +
+listmenutype main +
+deletemenu 1 +
+System deletes the new food item in index 1 from the menu. +
```
Comment (px1099) [inline docs/UserGuide.adoc]: Bad phrasing: "New food item"

**Review by px1099 — CHANGES_REQUESTED** [review]

**Comment by px1099** [comment]
So what does `viewallmenu` do?
It does not seem to be useful and user mistook it with `listmenu`

Or at least:
The key word `viewall` is used for viewing private details of a person in AB, and since `menu` don't hide its details, you should switch to the keyword `viewmenu`

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -94,28 +94,40 @@ Format: `listmenu`
...
+
+Example: +
+`listmenutype main` +
+`viewallmenu 1` +
+System shows 1st menu item of category type main.
+
 ==== Add Menu
 
 Add a new item to the menu. +
 Format: `addmenu NAME p/PRICE type/TYPE [t/TAG]`
```
Comment (px1099) [inline docs/UserGuide.adoc]: It is unclear that it allows multiple tags in the format

Comment (px1099) [reply docs/UserGuide.adoc]: From AB4 user guide:
Items with `…​` after them can be used multiple times including zero times e.g. `[t/TAG]…`​ can be used as   (i.e. 0 times), `t/friend`, `t/friend t/family`, etc.

**Comment by px1099** [comment]
You should add the 2 decimal digits restriction in the user guide

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -139,11 +151,14 @@ Format: `deletemenu INDEX`
...
 
 Example: +
-viewallmenu +
-deletemenu 1 +
-System deletes food item in index 1 from the menu. +
-findmenu Cheese Fries +
-deletemenu 3 +
+`listmenutype main` +
+`deletemenu 1` +
+System deletes menu item in index 1 from the menu. +
```
Comment (px1099) [inline docs/UserGuide.adoc]: Should be adjusted to:
... from the displayed menu of main dishes

File: docs/UserGuide.adoc [hunk 5]
```diff
@@ -139,11 +151,14 @@ Format: `deletemenu INDEX`
...
+`listmenutype main` +
+`deletemenu 1` +
+System deletes the menu item in index 1 from the menu. +
```
Comment (px1099) [inline docs/UserGuide.adoc]: Should be adjusted to:
... from the displayed menu of main dishes

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -139,11 +151,14 @@ Format: `deletemenu INDEX`
...
+`findmenu Cheese Fries` +
+`deletemenu 3` +
 System deletes Cheese Fries in index 3 from the menu.
```
Comment (px1099) [inline docs/UserGuide.adoc]: Should be adjusted to:
... from the displayed menu of items found with `Cheese Fries` keyword

File: docs/UserGuide.adoc [hunk 7]
```diff
@@ -152,12 +167,12 @@ Finds the menu item(s) related to the keywords typed in by the user. +
 Format: `findmenu KEYWORD [MORE KEYWORDS]`
 
 * A list of menu items with names related to the keyword(s) will be displayed.
-* The command is case sensitive.
+* The command is case insensitive.
```
Comment (px1099) [inline docs/UserGuide.adoc]: The matched results are case insensitive

**Comment by SalsabilTasnia** [comment]
> So what does `viewallmenu` do?
> It does not seem to be useful and user mistook it with `listmenu`
> 
> Or at least:
> The key word `viewall` is used for viewing private details of a person in AB, and since `menu` don't hide its details, you should switch to the keyword `viewmenu`

I am not changing this part since I am planning to get rid of `viewallmenu` command. I have not done it in this pr but I will do that in the following pr.

**Review by px1099 — APPROVED** [review]

**Pull request merged**